lib/inits/migrations/Migration.ts

```diff
@@ -1,27 +1,28 @@
 import { ManifestVersion } from "../../manifest/ManifestTypes"
 import { Effects } from "../../types"
+import { Utils } from "../../util"
 
-export class Migration<Version extends ManifestVersion> {
+export class Migration<WD, Version = ManifestVersion> {
   constructor(
     readonly options: {
       version: Version
-      up: (opts: { effects: Effects }) => Promise<void>
-      down: (opts: { effects: Effects }) => Promise<void>
+      up: (opts: { effects: Effects; utils: Utils<WD> }) => Promise<void>
+      down: (opts: { effects: Effects; utils: Utils<WD> }) => Promise<void>
     },
   ) {}
-  static of<Version extends ManifestVersion>(options: {
+  static of<WD, Version = ManifestVersion>(options: {
```

Comment:
Thought of doing a GDP for the WD and others like it to infer the type to let the user not put in generics, this might help with the 
```ts
A.of<WrapperData, "0.3.4.5">("0.3.4.5", ...rest)
```
And instead do something along the lines of 
```ts
A.of(WD, "0.3.4.5", ...rest)
```